### PR TITLE
feat: add MarkdownContent component to @backstage/core

### PR DESCRIPTION
**Pull request by jeskosda**
## Hey, I just made a Pull Request!

Since we are using a few internal plugins which are using the `ReactMarkdown` component with a specific style (influenced by the ReadMe Widget of https://github.com/RoadieHQ/backstage-plugin-github-insights), it may be useful to have a reusable component which is already preconfigured. 

### Screenshots
**Light Mode:**

Bildschirmfoto 2020-10-30 um 12 08 54

Bildschirmfoto 2020-10-30 um 12 09 11

**Dark-Mode:**

Bildschirmfoto 2020-10-30 um 12 09 02
Bildschirmfoto 2020-10-30 um 12 09 22

### CODE REVIEW

File: packages/theme/src/themes.ts [hunk 1]
```diff
@@ -48,6 +48,10 @@ export const lightTheme = createTheme({
       text: '#FFFFFF',
       link: '#000000',
     },
+    code: {
+      background: '#333333',
+      text: '#FFFFFF',
+    },
```
Comment (Fox32) [inline packages/theme/src/themes.ts:54]: I wonder if we can go with a similar styling like the code conponent? https://backstage.io/storybook/?path=/story/data-display-codesnippet--default

Comment (jeskosda) [reply packages/theme/src/themes.ts]: Like your idea! Added the CodeSnipped component as a custom renderer for code blocks. 

Bildschirmfoto 2020-10-30 um 14 25 11
Bildschirmfoto 2020-10-30 um 14 25 16

Comment (Fox32) [reply packages/theme/src/themes.ts]: Looks great

**Review by Fox32 — APPROVED** [review]
👍

**Review by stefanalund — APPROVED** [review]
Very nice, thanks!

(tests are failing, tried re-running them)

**Comment by freben** [comment]
Remaining error is actually codecov - needs more tests? :)

File: packages/core/src/components/MarkdownContent/MarkdownContent.test.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  it('render MarkdownContent component with common mark', async () => {
+    const rendered = await renderWithEffects(
+      wrapInTestApp(
+        <MarkdownContent content={'# H1\n' + '## H2\n' + '### H3'} />,
+      ),
+    );
+    expect(rendered.getByText('H1', { selector: 'h1' })).toBeInTheDocument();
+    expect(rendered.getByText('H2', { selector: 'h2' })).toBeInTheDocument();
+    expect(rendered.getByText('H3', { selector: 'h3' })).toBeInTheDocument();
+  });
```
Comment (Fox32) [inline packages/core/src/components/MarkdownContent/MarkdownContent.test.tsx:31]: @jeskosda  I think the coverage should be better if you add another test case for code blocks in markdown

**Comment by Fox32** [comment]
\o/ Yay, all green!

File: packages/core/src/components/MarkdownContent/MarkdownContent.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  enableGfm?: boolean;
+};
+
+const renderers = {
+  code: ({ language, value }: { language: string; value: string }) => {
+    return <CodeSnippet language={language} text={value} />;
+  },
+};
+
+export const MarkdownContent = ({ content, enableGfm = false }: Props) => {
```
Comment (andrewthauer) [inline packages/core/src/components/MarkdownContent/MarkdownContent.tsx:76]: Is there any reason to not enableGfm by default?

**Review by andrewthauer — COMMENTED** [review]
🎉

File: packages/core/src/components/MarkdownContent/MarkdownContent.tsx (hunk 3 above)
Comment (jeskosda) [reply packages/core/src/components/MarkdownContent/MarkdownContent.tsx:76]: One thought of mine was to have more control over when URLs are automatically converted to links. But basically the advantages outweigh the disadvantages. I am indifferent here. Could enable it by default.

Comment (andrewthauer) [reply packages/core/src/components/MarkdownContent/MarkdownContent.tsx:76]: Yeah, I also don't think it's a big deal either way. Just thinking most people probably assume GFM features as the default.

It looks like react-markdown uses remark 🎉, which I believe would be CommonMark by default. Can't recall if GFM is a 100% compatible superset or if there are some conflicting differences. On a side note, techdocs/mkdocs uses python-markdown which follows neither directly and doesn't behave the same for things like bulleted lists. The joys of competing markdown standards 😞.

File: packages/core/src/components/MarkdownContent/MarkdownContent.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    },
+  },
+}));
+
+/**
+ * MarkdownContent. Renders markdown (CommonMark, optionally with [GFM](https://github.com/remarkjs/remark-gfm)) to formatted HTML.
+ */
+type Props = {
+  content: string;
+  enableGfm?: boolean;
```
Comment (Rugvip) [inline packages/core/src/components/MarkdownContent/MarkdownContent.tsx:67]: Given the possible list of things here, should we make this `dialect="gfm"` instead? Agree that that could be default though, and one can use `dialect="common-mark"` if needed

Comment (andrewthauer) [reply packages/core/src/components/MarkdownContent/MarkdownContent.tsx:67]: Although, from the looks of it`enableGfm` maps to `plugins` which I'm taking as `remark` plugins which can add individual features rather then the dialect. I suppose `dialect` could be union with a future plugins later on though.

Comment (Rugvip) [reply packages/core/src/components/MarkdownContent/MarkdownContent.tsx:67]: Yeah setting `dialect="common-mark"` would translate to no plugins. I don't think we need a 1-to-1 mapping between dialects and remark plugins or anything like that. The dialect prop would just expose the possible dialects, and we may or may not stick to only those two forever.

Comment (jeskosda) [reply packages/core/src/components/MarkdownContent/MarkdownContent.tsx:67]: Thanks for the participation and the good ideas. I think it is not excluded that we invent our own dialects. A combination of different plugins could result into a new dialect. Adding the plugin [remark-emoji](https://github.com/rhysd/remark-emoji ) could result into the new dialect `"common-mark-emoji"` and so on...

I'm going to add this in this PR 👍

File: .changeset/3157.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+---
+'@backstage/core': minor
```
Comment (Rugvip) [inline .changeset/3157.md:2]: Just realized this'd be a patch change

**Review by Rugvip — APPROVED** [review]
Awesome stuff, thanks! :grin:

**Pull request merged**